lib/route-using-local-freerouting.ts

```diff
@@ -0,0 +1,157 @@
+import axios from "redaxios"
+import { readFileSync, existsSync } from "node:fs"
+import debug from "debug"
+import { DockerManager } from "./docker-manager"
+
+const log = debug("freerouting:route-using-local-freerouting")
+
+const handleApiError = (error: any) => {
+  console.error("API Error:")
+  if (error.data) {
+    console.error(`Status: ${error.status}`)
+    console.log("Response data:", error.data)
+  } else if (error.request) {
+    console.error("No response received from server")
+    console.error(error.request)
+  } else {
+    console.error("Error:", error.message)
+  }
+  throw error
+}
+
+interface RouteOptions {
+  inputPath: string
+  port?: number
+}
+
+export async function routeUsingLocalFreerouting({
+  inputPath,
+  port = 37864,
+}: RouteOptions): Promise<string> {
+  if (!existsSync(inputPath)) {
+    throw new Error(`Input file does not exist: ${inputPath}`)
+  }
+
+  const dockerManager = new DockerManager(port)
+  const API_BASE = `http://localhost:${port}`
+  const headers = {
+    "Freerouting-Profile-ID": "e9866fac-e7ae-4f9f-a616-24ec577aa461",
+    "Freerouting-Environment-Host": "tscircuit/0.0.1",
+  }
+
+  try {
+    // Stop any existing container before starting a new one
+    await dockerManager.stopContainer()
```

Comment:
how can you stop a container if one isn't running? inside the containerId will be null